platform/darwin/src/MGLTilePyramidOfflineRegion.mm

```diff
@@ -53,6 +53,11 @@ - (instancetype)initWithStyleURL:(NSURL *)styleURL bounds:(MGLCoordinateBounds)b
     return self;
 }
 
+-(uint64_t)tileCount {
+    auto tilePyramidOfflineRegion = [self offlineRegionDefinition];
+    return tilePyramidOfflineRegion.tileCount(mbgl::style::SourceType::Vector, 512, {static_cast<unsigned char>(tilePyramidOfflineRegion.minZoom), static_cast<unsigned char>(tilePyramidOfflineRegion.maxZoom)});
```

Comment:
For testing purposes, I'm assuming the source type and tile size. We will need to figure out a way for to pass these in dynamically, though.

Reply:
The offline map API and runtime styling APIs currently have little to do with each other. I think it would be problematic if we hooked up MGLOfflinePack or MGLOfflineRegion with MGLStyle in any way, because that would imply you could add a source to the style and have it reflected in the tile count, or change a layer in the style and persist that change when loading the offline pack in the future.

For now, let’s turn the `MGLOfflineRegion.tileCount` property into a method that takes an MGLTileSource as a parameter. We can frame it as a way to get the tile count for a single source _within_ the offline pack: `-countOfTilesInSource:`. Then the developer can create an MGLTileSource to pass into this method. The MGLTileSource knows all the details we need for that `tileCount()` call even without loading the style.

It’ll be a bit weird that the developer will be able to pass an unrelated source into `-countOfTilesInSource:`, but that weirdness comes from how the style URL is part of the offline region, something we can’t change just yet.